File: poc-kronos-frontend/src/model/basecurricular/Competencia.js
class Competencia {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idNivel = dados ? dados.idNivel : null;
    this.nivel = dados ? dados.nivel : null;
    this.idComponente = dados ? dados.idComponente : null;
    this.componente = dados ? dados.componente : null;
    this.descricao = dados ? dados.descricao : null;
    this.codigo = dados ? dados.codigo : null;
    this.ativo = dados ? dados.ativo : null;
    this.bncc = dados ? dados.bncc : null;
    this.geral = dados ? dados.geral : null;
  }

  static validations(required, maxLength) {
    return {
      descricao: { required, maxLength: maxLength(500) },
      codigo: { required, maxLength: maxLength(10) },
      idNivel: { required }
    };
  }

  static validationMessages() {
    return {
      descricao: {
        required: "Descrição é obrigatória",
        maxLength: "Descrição só aceita no máximo 500 caracteres"
      },
      codigo: {
        required: "Código é obrigatório",
        maxLength: "Código só aceita no máximo 10 caracteres"
      },
      idNivel: {
        required: "Nível é obrigatório"
      }
    };
  }
}
export default Competencia;

File: src/main/java/br/com/kronos/backend/aplicacao/basecurricular/Competencia.java
package br.com.kronos.backend.aplicacao.basecurricular;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Competencia {
    
    private long id;
    private String descricao;
    private Long idNivel;
    private Long idComponente;
    private String codigo;
    private boolean ativo;
    private Boolean bncc;
    private Boolean geral;
    private Long idInstituicao;

	@Builder
    public Competencia(long id, String descricao, Long idNivel, Long idComponente, String codigo, 
                       boolean ativo, Boolean bncc, Boolean geral, Long idInstituicao) {

		super();
        this.id = id;
        this.descricao = exigirTamanho(descricao, "Descrição", 1, 500);
        this.idNivel = exigirNaoNulo(idNivel, "Nível");    
        
        this.codigo = exigirTamanho(codigo, "Código", 1, 10);
        this.ativo = exigirNaoNulo(ativo, "Ativo");
        if (bncc != null && bncc) {
        	this.bncc = bncc;
        	this.idInstituicao = null; 
        } else {
        	this.bncc = false;
        	this.idInstituicao = idInstituicao;     
        }
        if (geral != null && geral) {
        	this.geral = geral;
        	this.idComponente = null;  
        } else {
        	this.geral = false;
        	this.idComponente = idComponente;  
        }
	}
}
